lib/index.ts

```diff
@@ -463,6 +482,41 @@ export class DockerProgress {
 		return hash;
 	}
 
+	/** Build docker image calling onProgress with extended progress info regularly */
+	async build(
+		tarStream: Stream.Readable,
+		onProgress: ProgressCallback,
+		options?: BuildOptions,
+	): Promise<string> {
+		const ignoreErrorEvents = !!options?.ignoreProgressErrorEvents;
+
+		// Authconfig is not supported by docker `/build` endpoint but we support
+		// it here to provide a common interface with the `pull()` method
+		if (options?.authconfig) {
+			const { serveraddress, ...authconfig } = options.authconfig;
+			options.registryconfig = {
+				[serveraddress]: authconfig,
+
+				// Override with registryconfig if it the same serveraddress
+				// is used in both options
+				...(options.registryconfig ?? {}),
+			};
+
+			delete options.authconfig;
+		}
```

Comment:
What do you think about this @pdcastro? Is mostly a convenience conversion to make the options of the pull and build methods more compatible.

Reply:
I am happy with it. :-)